src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -517,6 +520,14 @@ int32_t CCARTree::compute_best_attribute(SGMatrix<float64_t> mat, SGVector<float
 			}
 		}
 	}
+	
+	SGVector<index_t> idx(num_feats);
+	idx.range_fill();
+	if (subset_size)
+	{
+		num_feats=subset_size;
+		CMath::permute(idx);
```

Comment:
Note that if you permute here and then later loop over idx, then your memory access will be random.